tests/core/generic-device-reg/task.yaml

```diff
@@ -0,0 +1,66 @@
+summary: |
+    Test that device initialisation and registration work with the model
+    specifying serial-authority: [generic]
+
+# TODO:UC20: enable for UC20, it assumes /var/lib/snapd/seed/assertions/model
+#            which we don't have currently
+systems: [ubuntu-core-18*]
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    systemctl stop snapd.service snapd.socket
+    rm -rf /var/lib/snapd/assertions/*
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+    mv /var/lib/snapd/seed/assertions/model model.bak
+    # get the account and signing account-key of the
+    # pc-18-amd64-accept-generic.model's signer (pedronis)
+    snap known --remote account account-id=CA5GLZgNQWPhspDQK63Er46Uxz2SO7ez > /var/lib/snapd/seed/assertions/pedronis.account
+    snap known --remote account-key public-key-sha3-384=x1Tnl94nkgb-rMDs_l63gvgkLYGjXKfpkCCIQZwJ72-LR6X6OrSvNS9z2WS1lAGz > /var/lib/snapd/seed/assertions/test-models.account-key
+    cp "$TESTSLIB"/assertions/pc-18-amd64-accept-generic.model /var/lib/snapd/seed/assertions/
+
+    # kick first boot again
+    systemctl start snapd.service snapd.socket
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    systemctl stop snapd.service snapd.socket
+    rm -rf /var/lib/snapd/assertions/*
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+    rm -f /var/lib/snapd/seed/assertions/pedronis.account
+    rm -f /var/lib/snapd/seed/assertions/test-models.account-key
+    rm -f /var/lib/snapd/seed/assertions/pc-18-amd64-accept-generic.model
+    cp model.bak /var/lib/snapd/seed/assertions/model
+    # kick first boot again
+    systemctl start snapd.service snapd.socket
```

Comment:
Is it necessary for restore to do this?

Reply:
I'm not sure actually, historically all the tests about firstboot restore this way